### PR TITLE
Second incremental speed up of Mitiq tests

### LINKED ISSUE #1998: Speed up Mitiq tests
## Issue Description

This is a continuation of the progress made on speeding up Mitiq's tests in milestone 0.29. 
previous issue: https://github.com/unitaryfund/mitiq/issues/1427
first PR speeding up tests: https://github.com/unitaryfund/mitiq/pull/1996

## Proposed Solution

Remove redundant tests, and look for other opportunities to reduce test time.

### PR DESCRIPTION
**Pull request by Misty-W**
## Description

Fixes #1998.
Speed up tests by:
1. Setting `force_run_all` in most PEC tests to `False` (except where already explicitly set to `True`). 
2. Reduce `num_training_circuits` in CDR tests from 20 to 10. 
3. Remove sample_bitstrings executor test case from for `test_with_observable_two_qubits()`, since `execute_with_zne` with `sample_bitstrings` is already tested elsewhere
4. Replace the `str` typed bitstrings with `int` in `prepare_logical_0_state_for_5_1_3_code()` of`test_qse.py`

Comparing on CI: on average the tests ran ~28.5~ 25.5  minutes on this PR  vs. ~33~ 31.1 minutes on latest run of `master`.
---

### License

- [x] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.
- [x] I added unit tests for new code.
- [x] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [x] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [x] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.
- [x] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

**Comment by Misty-W** [comment]
No change to docs. Only tests. 

RTD shows failed in the list of checks, but the build completed successfully, it's just that GitHub displays anything with >1h build time as a fail (which it is in a way, just not to be fixed in this PR).

**Comment by purva-thakre** [comment]
> That needs to be addressed also, but it's outside the scope of this PR.

Yep, I have noticed this RTD build failure in other PRs also.

**Comment by natestemen** [comment]
Just as we did in our [previous PR](https://github.com/unitaryfund/mitiq/pull/1996), can you post the before and after duration of the tests?

**Comment by Misty-W** [comment]
Note to self: check tests listed in https://github.com/unitaryfund/mitiq/issues/1427 .

**Comment by Misty-W** [comment]
@natestemen ~5.5 minute improvement on average over latest run of `master` 🥳 

Personally I think it's enough improvement to merge for this milestone, and of course we'll keep improving in future milestones.

File: mitiq/qse/tests/test_qse.py [hunk 1]
```diff
@@ -317,38 +317,21 @@ def gram_schmidt(
...
-        "10011",
-        "01000",
-    ]:
-        logical_1_state[int(z, 2)] = -1 / 4
+    # bitstrings ["11111", "01101", "10110", "01011", "10101", "11010"]
+    np.put(logical_1_state, [11, 13, 21, 22, 26, 31], 1 / 4)
+    # bitstrings: [
+    #  "00100", "11001", "00111", "00010", "11100",
+    #  "00001", "10000", "01110", "10011", "01000"]
+    np.put(logical_1_state, [1, 2, 4, 7, 8, 14, 16, 19, 25, 28], 1 / 4)
```
Comment (natestemen) [inline mitiq/qse/tests/test_qse.py:334]: Huh TIL `numpy.put`! I think the second instance here should be `-1 / 4`? Also can we keep the order of the bitstrings and the converted int in a consistent order? 

As far as speeding things up, I can't imagine this has much of an impact, does it?

**Review by natestemen — COMMENTED** [review]
From local testing (only one round of `make test`):

| before | after |
| ------ | ----- |
| 6:27   | 5:38  |

If only CI would run this fast!

File: mitiq/calibration/settings.py [hunk 2]
```diff
@@ -473,6 +473,7 @@ def make_strategies(self) -> List[Strategy]:
             "is_qubit_dependent": False,
             "noise_level": 0.001,
             "num_samples": 200,
+            "force_run_all": False,
```
[the remark this answers is not in the document]
Comment (Misty-W) [reply mitiq/calibration/settings.py:476]: Yes and yes (by about ~1-2 min).

File: mitiq/qse/tests/test_qse.py (hunk 1 above)
Comment (Misty-W) [reply mitiq/qse/tests/test_qse.py]: It did improve CI by ~1 min on average, so I decided to fix the missing `-` sign and make the ordering consistent after all.

**Review by natestemen — APPROVED** [review]

**Pull request merged**